com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationsCloudModelBuilder.java

```diff
@@ -135,15 +140,56 @@ protected CloudApplicationExtended createCloudApplication(String name, String mo
         return app;
     }
 
-    protected void warnAboutUnsupportedParameters(List<Map<String, Object>> fullParametersList) {
+    protected void validateParameters(List<Map<String, Object>> fullParametersList) {
         Map<String, Object> merged = mergeProperties(fullParametersList);
-        applicationEnvCloudModelBuilder.removeSpecialApplicationProperties(merged);
-        applicationEnvCloudModelBuilder.removeSpecialServiceProperties(merged);
-        for (String parameterName : merged.keySet()) {
+
+        warnAboutDeprecatedParameters(merged);
+        warnAboutUnsupportedParameters(merged);
+    }
+
+    protected void warnAboutUnsupportedParameters(Map<String, Object> parametersInApp) {
+        applicationEnvCloudModelBuilder.removeSpecialApplicationProperties(parametersInApp);
+        applicationEnvCloudModelBuilder.removeSpecialServiceProperties(parametersInApp);
+        for (String parameterName : parametersInApp.keySet()) {
             LOGGER.warn(MessageFormat.format(Messages.UNSUPPORTED_PARAMETER, parameterName));
         }
     }
 
+    protected void warnAboutDeprecatedParameters(Map<String, Object> parametersInApp) {
+        List<Parameter> deprecatedParams = filterDeprecatedParameters(parametersInApp.keySet());
+
+        if (deprecatedParams != null && !deprecatedParams.isEmpty()) {
+
+            String combinedParameterNames = deprecatedParams.stream()
+                .map(Parameter::getName)
+                .collect(Collectors.joining(", "));
+
+            String combinedCustomDeprecationMessage = deprecatedParams.stream()
+                .filter(p -> p.getSpecificDeprecationMessage() != null)
+                .map(Parameter::getSpecificDeprecationMessage)
+                .collect(Collectors.joining(" "));
+            // TODO: check why this is even nullable
+            if (userMessageLogger != null) {
+                userMessageLogger.info(Messages.GENERAL_DEPRECATED_PARAMETER, combinedParameterNames);
+                if (combinedCustomDeprecationMessage != null) {
+                    userMessageLogger.info(combinedCustomDeprecationMessage);
+                }
+            }
+        }
+    }
+
+    protected List<Parameter> filterDeprecatedParameters(Collection<String> parameterNames) {

```

Comment:
`filter` is a bit ambiguous. Does it return the deprecated or the non-deprecated parameters?